core/src/main/scala/io/chrisdavenport/rediculous/RedisConnection.scala

```diff
@@ -86,6 +84,13 @@ object RedisConnection{
     }
   })
 
+  private[rediculous] def closeReturn[F[_]: MonadError[*[_], Throwable], A](fE: F[Either[Resp, A]]): F[A] = 
+    fE.flatMap{
+        case Right(a) => a.pure[F]
+        case Left(e@Resp.Error(_)) => ApplicativeError[F, Throwable].raiseError[A](e)
+        case Left(other) => ApplicativeError[F, Throwable].raiseError[A](new Throwable(s"Rediculous: Incompatible Return Type: Got $other"))
```

Comment:
⛏️ As a consumer I really like it when all errors raised from the library inherit a common base type

Reply:
Yeah. I could make some general type at the top level and have others inherit that. Wouldn't be matchable. I'll do that in follow up PR though as I don't want to confuse this one with the error manipulation